railties/lib/rails/generators/app_base.rb

```diff
@@ -564,22 +564,12 @@ def dockerfile_deploy_packages
         packages.compact.sort
       end
 
-      # CSS processors other than Tailwind and Sass require a node-based JavaScript environment. So overwrite the normal JS default
-      # if one such processor has been specified.
-      def adjusted_javascript_option
-        if options[:css] && options[:css] != "tailwind" && options[:css] != "sass" && options[:javascript] == "importmap"
-          "esbuild"
-        else
-          options[:javascript]
-        end
-      end
-
       def css_gemfile_entry
         return unless options[:css]
 
-        if !using_node? && options[:css] == "tailwind"
+        if options[:css] == "tailwind" && options[:javascript] == "importmap"
```

Comment:
Look good for me! I have used form `options[:javascript] == "importmap"` because it has been a literal translation of the PR table, but it seems to me a successful change.